mcp/mcp-server/src/main/java/software/amazon/smithy/java/mcp/server/McpService.java

```diff
@@ -112,6 +115,52 @@ public JsonRpcResponse handleRequest(
             Consumer<JsonRpcResponse> asyncResponseCallback,
             ProtocolVersion protocolVersion
     ) {
+        if (metricsObserver != null) {
+            try {
+                if ("initialize".equals(req.getMethod())) {
```

Comment:
You don't need to do any matching here. We already do it below just add these to the methods there.